pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1175,6 +1175,11 @@ func (p *PriorityQueue) movePodsToActiveOrBackoffQueue(logger klog.Logger, podIn
 
 	activated := false
 	for _, pInfo := range podInfoList {
+		// Since there may be many gated pods and they will not move from the
+		// unschedulable pool, we skip calling the expensive isPodWorthRequeueing.
+		if pInfo.Gated {
+			continue
```

Comment:
We're trying to have requeueing logs with level:5.

```suggestion
			logger.V(5).Info("Event is not making the pod schedulable because the pod is gated", "pod", klog.KObj(pInfo.Pod), "event", event.Label))
			continue
```

Reply:
Can we not? Just processing the arguments of the function can make a difference in the performance.

Reply:
Not having logs made it hard for me to figure out that this if check caused pods to get stuck. See https://github.com/kubernetes/kubernetes/issues/125223#issuecomment-2170541032

Reverting this change here fixed https://github.com/kubernetes/kubernetes/issues/125223, at least for the test case that I have been experimenting with. I suppose the others will now also work again.